src/main/java/sx/blah/discord/api/internal/DiscordClientImpl.java

```diff
@@ -112,6 +113,11 @@
 	 * The specific shard (if there is one) that the client is running on.
 	 */
 	private final int[] shard;
+	
+	/**
+	 * The component registry singleton for this client.
+	 */
+	private final ComponentRegistry componentRegistry = new ComponentRegistry();
```

Comment:
clients should have their own because components registered to registries should not be global